Warnings in sockets provider

I see these when compiling with gcc 4.9.1 on RHEL 6.5 -- they feel like real errors:

```
  CC       prov/sockets/src/sock_dgram.lo
../prov/sockets/src/sock_dgram.c:730:2: warning: initialization from incompatible pointer type
  .tx_ctx = sockd_ep_tx_ctx,
  ^
../prov/sockets/src/sock_dgram.c:730:2: warning: (near initialization for 'sockd_ops_ep.tx_ctx')
../prov/sockets/src/sock_dgram.c:731:2: warning: initialization from incompatible pointer type
  .rx_ctx = sockd_ep_rx_ctx,
  ^
../prov/sockets/src/sock_dgram.c:731:2: warning: (near initialization for 'sockd_ops_ep.rx_ctx')

...

  CC       prov/sockets/src/sock_rdm.lo
../prov/sockets/src/sock_rdm.c:1355:2: warning: initialization from incompatible pointer type
  .tx_ctx = sock_rdm_ep_tx_ctx,
  ^
../prov/sockets/src/sock_rdm.c:1355:2: warning: (near initialization for 'sock_rdm_ep_ops.tx_ctx')
../prov/sockets/src/sock_rdm.c:1356:2: warning: initialization from incompatible pointer type
  .rx_ctx = sock_rdm_ep_rx_ctx,
  ^
../prov/sockets/src/sock_rdm.c:1356:2: warning: (near initialization for 'sock_rdm_ep_ops.rx_ctx')
```

I'm not sure who is responsible for the sockets provider, but I thought I'd report these nonetheless.

--- a/prov/sockets/src/sock_dgram.c
+++ b/prov/sockets/src/sock_dgram.c
@@ -514,7 +514,7 @@ static int sockd_ep_setopt(fid_t fid, int level, int optname,
 	return -errno;
 }
 
-static int sockd_ep_tx_ctx(struct fid_ep *ep, int index,
+static int sockd_ep_tx_ctx(struct fid_sep *sep, int index,
 		struct fi_tx_ctx_attr *attr, struct fid_ep **tx_ep,
 		void *context)
 {
@@ -523,7 +523,7 @@ static int sockd_ep_tx_ctx(struct fid_ep *ep, int index,
 }
 
 
-static int	sockd_ep_rx_ctx(struct fid_ep *ep, int index,
+static int sockd_ep_rx_ctx(struct fid_sep *sep, int index,
 			struct fi_rx_ctx_attr *attr, struct fid_ep **rx_ep,
 			void *context)
 {

--- a/prov/sockets/src/sock_rdm.c
+++ b/prov/sockets/src/sock_rdm.c
@@ -1280,13 +1280,13 @@ int sock_rdm_ep_setopt(fid_t fid, int level, int optname,
 	return 0;
 }
 
-int sock_rdm_ep_tx_ctx(struct fid_ep *ep, int index, struct fi_tx_ctx_attr *attr, 
-		    struct fid_ep **tx_ep, void *context)
+int sock_rdm_ep_tx_ctx(struct fid_sep *sep, int index, struct fi_tx_ctx_attr *attr, 
+		       struct fid_ep **tx_ep, void *context)
 {
 	struct sock_ep *sock_ep;
 	struct sock_tx_ctx *tx_ctx;
 
-	sock_ep = container_of(ep, struct sock_ep, ep.fid);
+	sock_ep = container_of(sep, struct sock_ep, ep.fid);
 	if (index >= sock_ep->ep_attr.tx_ctx_cnt)
 		return -FI_EINVAL;
 
@@ -1313,13 +1313,13 @@ int sock_rdm_ep_tx_ctx(struct fid_ep *ep, int index, struct fi_tx_ctx_attr *attr
 	return 0;
 }
 
-int sock_rdm_ep_rx_ctx(struct fid_ep *ep, int index, struct fi_rx_ctx_attr *attr,
-		    struct fid_ep **rx_ep, void *context)
+int sock_rdm_ep_rx_ctx(struct fid_sep *sep, int index, struct fi_rx_ctx_attr *attr,
+		       struct fid_ep **rx_ep, void *context)
 {
 	struct sock_ep *sock_ep;
 	struct sock_rx_ctx *rx_ctx;
 
-	sock_ep = container_of(ep, struct sock_ep, ep.fid);
+	sock_ep = container_of(sep, struct sock_ep, ep.fid);
 	if (index >= sock_ep->ep_attr.rx_ctx_cnt)
 		return -FI_EINVAL;
 

